template.mustache

```diff
@@ -0,0 +1,93 @@
+%global npm_name {{NAME}}
+%global enable_tests 0
+
+Name: nodejs-%{npm_name}
+Version: {{VERSION}}
+Release: {{RELEASE}}%{?dist}
+Summary: {{SUMMARY}}
+License: {{LICENSETYPE}}
+Group: Development/Libraries
+URL: {{&PROJECTURL}}
+{{#SOURCES}}
+{{&.}}
+{{/SOURCES}}
+BuildRequires: nodejs-packaging
+BuildArch:  noarch
+ExclusiveArch: %{nodejs_arches} noarch
+
+{{#BUNDLED}}
+Provides: npm(%{npm_name}) = %{version}
+{{#PROVIDES}}
+Provides: bundled-npm({{name}}) = {{version}}
+{{/PROVIDES}}
+AutoReq: no
+AutoProv: no
+
+%define npm_cache_dir /tmp/npm_cache_%{name}-%{version}-%{release}
+{{/BUNDLED}}
+{{^BUNDLED}}
+{{#DEPENDENCIES}}
+{{&.}}
+{{/DEPENDENCIES}}
+{{/BUNDLED}}
+
+%description
+%{summary}
+
+%prep
+{{#BUNDLED}}
+mkdir -p %{npm_cache_dir}
+for tgz in %{sources}; do
+  echo $tgz | grep -q registry.npmjs.org || npm cache add --cache %{npm_cache_dir} $tgz
+done
+%setup -T -q -a {{PROVIDES.length}} -D -n %{npm_cache_dir}
```

Comment:
I'm not 100% sure about the PROVIDES.length here or if there'll be an off-by-one error.